server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -818,7 +818,8 @@ public void publish(ClusterChangedEvent clusterChangedEvent, ActionListener<Void
                 // there is no equals on cluster state, so we just serialize it to XContent and compare JSON representation
                 assert clusterChangedEvent.previousState() == coordinationState.get().getLastAcceptedState() ||
                     Strings.toString(clusterChangedEvent.previousState()).equals(
-                        Strings.toString(clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState())));
+                        Strings.toString(clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState())))
+                    : clusterChangedEvent.previousState() + " vs " + clusterStateWithNoMasterBlock(coordinationState.get().getLastAcceptedState());
```

Comment:
This assertion failed at some point, unreproducibly. If it fails again then this should help.